data_diff/databases/postgresql.py

```diff
@@ -168,6 +168,7 @@ def create_connection(self):
 
         pg = import_postgresql()
         try:
+            self._args["password"] = unquote(self._args["password"])
```

Comment:
How are you passing in this parameter when connecting? I'm able to use a password with special characters just fine when connecting using the `connect_to_table` method with a dictionary. This test case passes when I run it without your changes. I recommend use this method instead of constructing the URL connection string. Let me know if I'm missing something! 

```python
class TestSpecialCharacterPassword(unittest.TestCase):
    def setUp(self) -> None:
        self.connection = get_conn(db.PostgreSQL)

        table_suffix = random_table_suffix()

        self.table_name = f"table{table_suffix}"
        self.table = table(self.table_name)

    def test_special_char_password(self):
        password = "passw!!!@rd"
        # Setup user with special character '@' in password
        self.connection.query("DROP USER IF EXISTS test;", None)
        self.connection.query(f"CREATE USER test WITH PASSWORD '{password}';", None)
        
        connect_to_table(
            {
                "driver": "postgresql",
                "host": "localhost",
                "port": 5432,
                "dbname": "postgres",
                "user": "test",
                "password": password,
            },
            self.table_name,
        )
```

Reply:
Thx for replay. As you said, If you pass a dict to `connect_to_table` function, there is nothing wrong. But when passing a url with special characters , connection will failed because of dns wrong parsing. I think that since we allow `connect_to_table` to pass in the url string as a parameter, we must also deal with the possible consequences of url errors.

Reply:
for example, If i write codes like this: `db_url = f"postgresql://{config['user']}:{config['password']}@{config['host']}:{config['port']}/{config['database']}"`. And these parameters are from out conf files,  when the `config['password']` has special characters like `@`, dns parse will failed.

Reply:
Thanks for providing the example url construction code. Fair point, and verified your fix works. Can you add the below unit test to your branch? 

```python
class TestSpecialCharacterPassword(unittest.TestCase):
    def setUp(self) -> None:
        self.connection = get_conn(db.PostgreSQL)

        table_suffix = random_table_suffix()

        self.table_name = f"table{table_suffix}"
        self.table = table(self.table_name)

    def test_special_char_password(self):
        password = "passw!!!@rd"
        # Setup user with special character '@' in password
        self.connection.query("DROP USER IF EXISTS test;", None)
        self.connection.query(f"CREATE USER test WITH PASSWORD '{password}';", None)

        password_quoted = quote(password)
        db_config = {
                "driver": "postgresql",
                "host": "localhost",
                "port": 5432,
                "dbname": "postgres",
                "user": "test",
                "password": password_quoted,
            }

        # verify pythonic connection method
        connect_to_table(
            db_config,
            self.table_name,
        )

        # verify connection method with URL string unquoted after it's verified
        db_url = f"postgresql://{db_config['user']}:{db_config['password']}@{db_config['host']}:{db_config['port']}/{db_config['dbname']}"

        connection_verified = connect(db_url)
        assert connection_verified._args.get('password') == password
```

Reply:
Here are the updated imports.

```python
import unittest

from urllib.parse import quote
from data_diff.queries.api import table, commit
from data_diff import TableSegment, HashDiffer
from data_diff import databases as db
from tests.common import get_conn, random_table_suffix, connect
from data_diff import connect_to_table
```

Reply:
Already add UT in test_postgresql.py. Thanks for providing the codes.